dynamic_sampling_context.go

```diff
@@ -16,7 +17,7 @@ type DynamicSamplingContext struct {
 	Frozen  bool
 }
 
-func NewDynamicSamplingContext(header []byte) (DynamicSamplingContext, error) {
+func NewDynamicSamplingContextFromHeader(header []byte) (DynamicSamplingContext, error) {
```

Comment:
I'd use `DynamicSamplingContextFromHeader` instead. `NewX` should be reserved for direct builders.